telegram/ext/_dispatcher.py

```diff
@@ -482,7 +479,7 @@ def process_update(self, update: object) -> None:
         handled = False
         sync_modes = []
 
-        for group in self.groups:
+        for group in sorted(self.handlers):  # lower -> higher groups
```

Comment:
sorting on every `process_update` seems like a bad idea, even if one usually only has a small number of groups. Let's sort `self.handlers` in `add_handler` - maybe we can even just insert new groups directly in the right order? 🤔